src/GraphQL/Instrumentation/FieldResolverBuilderExtensions.cs

```diff
@@ -35,9 +35,9 @@ public static IFieldMiddlewareBuilder Use(this IFieldMiddlewareBuilder builder,
                 }
 
                 var parameters = methodInfo.GetParameters();
-                if (parameters.Length != 2 || parameters[0].ParameterType != typeof(ResolveFieldContext) || parameters[1].ParameterType != typeof(FieldMiddlewareDelegate))
+                if (parameters.Length != 2 || parameters[0].ParameterType != typeof(IResolveFieldContext) || parameters[1].ParameterType != typeof(FieldMiddlewareDelegate))
```

Comment:
Most of the time I spent thinking about whether it is right to put the inequality operator here - see changed condition in `ReflectionHelper.BuildArguments`. In principle, it’s correct, but some doubts remain.

Reply:
I'd leave it as-is for this PR.  It could be "enhanced" to match on a typed IResolveFieldContext and then only run the middleware if the context type matches.  I probably wouldn't do that, though.